### DISCUSSION TITLE
nvidia jetpack rtl8812au monitor no capture

### DISCUSSION THREAD

**Discussion by amtcvx** [svpcom/wfb-ng#127] [Questions]
Hello, 

I am running wifibroadcast successfull with rt2800usb and 88XXau, on raspberry (Raspian Buster) pi0, pi3B+, pi4, PC desktop (Ubuntu 1804), and Nvidia Nano, Xavier NX (Jetpack 4.5.0).

However, for Nvidias with 88XXau the wlan0 interface do not receive messages with svpcom/rtl8812au/v5.6.4.2.
while there are received with aircrack-ng/rtl8812au/v5.7.0 (5.9.3.2).
But messages are not more recognized by wifibroadcast with version
rx.cpp => "unknown encapsulation"

How can I use wifibroadcast with rtl8812au on the NVIDIA boards ?

Thank you for your support.
XP

**Comment by svpcom** [comment]
See https://github.com/svpcom/wifibroadcast/discussions/123
Remove rtl8812au.ko driver

**Reply by amtcvx** [reply]
Hello

I confirm that the original driver rtl8812au is blacklisted (blacklist.conf), and that I am using the one installed with dkms_install.

ethtool -i wlan0
=> driver: rtl88XXau

lsmod
Module                  Size  Used by
88XXau               2042778  0
cfg80211              589932  1 88XXau

On PC desktop (Ubuntu 1804) tcpdump show messages sent & receive
On NVIDIA boards tcpdump show only messages sent, and nothing is received

Thank you for your help
XP

**Reply by svpcom** [reply]
1. What kernel version do you use?
2. Check that CRDA is disabled or you use channel allowed by CRDA (iw reg get)
3. If you use rt2800usb on Nvidia board and rtl8812au on RPI - is it works?
4. What rtl8812au card do you use?
5. With 5.9.3.2 (aircrack-ng driver) does tcpdump show incoming packets? If yes, please dump them (using -w) and attach the dump. Also please add `printf("encap_type=%d\n", link_encap);` after `int link_encap = pcap_datalink(ppcap);` in `src/rx.cpp`

**Reply by amtcvx** [reply]
Hello

1) NVIDIA boards with Jetpack 4.5.0, kernel 4.9

2) 
iw reg get
global
country DE: DFS-ETSI
	(2400 - 2483 @ 40), (N/A, 20), (N/A)
	(5150 - 5250 @ 80), (N/A, 20), (N/A), NO-OUTDOOR, AUTO-BW
	(5250 - 5350 @ 80), (N/A, 20), (0 ms), NO-OUTDOOR, DFS, AUTO-BW
	(5470 - 5725 @ 160), (N/A, 26), (0 ms), DFS
	(5725 - 5875 @ 80), (N/A, 13), (N/A)
	(57000 - 66000 @ 2160), (N/A, 40), (N/A)

iwconfig
wlan0     unassociated  ESSID:""  Nickname:"<WIFI@REALTEK>"
          Mode:Monitor  Frequency=5.2 GHz  Access Point: Not-Associated   

3) Yes it is works:
NVIDIA with rt2800usb <-> PC desktop (ubuntu 18.04) with rt2800usb or 88XXau

4) ASUS USB-AC56 and https://imi-link.com/portfolio/1200mbps-11ac-usb-wireless-receiver

5) 5.9.3.2 (aircrack-ng driver) on both sides: NVIDIA and PC desktop
git clone -b v5.7.0 https://github.com/aircrack-ng/rtl8812au.git
sudo make dkms_install
=> /lib/modules/4.9.201-tegra/updates/dkms/88XXau.ko
=> /lib/modules/4.15.0-43-generic/updates/dkms/88XXau.k

 unknown encapsulation => 105
 (DLT_IEEE802_11 /* IEEE 802.11 wireless */ ?)

 Thank you for your support
 XP

**Reply by svpcom** [reply]
1. What WiFi channel do you use?
2. Could you attach full logs of wifibroadcast with 8812au card?
3. What hardware (card model) do you use?

It seem that card or driver doesn't support/enter into monitor mode. Try following steps manually and check for errors:
```
ifconfig $WLAN down
iw dev $WLAN set monitor otherbss
iw reg set BO
ifconfig $WLAN up
iw dev $WLAN set channel $CHANNEL5G
./wfb_rx -p 1 -u 5600 -K gs_or_drone.key $WLAN
```

**Reply by amtcvx** [reply]
Hello,

1) channel 40
2) I only use ./wfb_tx and ./wfb_rx
3) Asus USB-AC56 on Nano Dev kit or Xavier Nx on connecttech Quark Carrier board

ifconfig wlan0 down
iw dev wlan0 set monitor otherbss
iw reg set BO
ifconfig wlan0 up
iw dev wlan0 set channel 40

No error
 
iwconfig
=>
wlan0     unassociated  ESSID:""  Nickname:"<WIFI@REALTEK>"
          Mode:Monitor  Frequency=5.2 GHz  Access Point: Not-Associated   
          Sensitivity:0/0  
          Retry:off   RTS thr:off   Fragment thr:off
          Power Management:off
          Link Quality:0  Signal level:0  Noise level:0
          Rx invalid nwid:0  Rx invalid crypt:0  Rx invalid frag:0
          Tx excessive retries:0  Invalid misc:0   Missed beacon:0


iw wlan0 info
=>
Interface wlan0
	ifindex 8
	wdev 0x100000001
	addr 24:4b:fe:b7:26:18
	type monitor
	wiphy 1
	txpower 20.00 dBm

modinfo 88XXau 
(joined file)
[modinfo_88XXau.txt](https://github.com/svpcom/wifibroadcast/files/6190002/modinfo_88XXau.txt)

**Reply by amtcvx** [reply]
hcxdumptool -i wlan0 --check_driver
=> 
initialization...
starting driver test...
warning possible interfere: NetworkManager is running with pid 4274

warning possible interfere: wpa_supplicant is running with pid 4275

driver tests passed...
all required ioctl() system calls are supported by driver

terminating...


hcxdumptool -i wlan0 --check_injection
=>
initialization...
warning possible interfere: NetworkManager is running with pid 4274

warning possible interfere: wpa_supplicant is running with pid 4275

starting antenna test and packet injection test (that can take up to two minutes)...
available channels: 1,2,3,4,...,255
packet injection is working on 2.4GHz!
packet injection is working on 5GHz!
injection ratio: 40% (BEACON: 88 PROBERESPONSE: 36)
your injection ratio is average, but there is still room for improvement
antenna ratio: 27% (NETWORK: 18 PROBERESPONSE: 5)
your antenna ratio is average, but there is still room for improvement

terminating...

**Reply by svpcom** [reply]
1. You need to disable NetworkManager and wpa_supplicant for rlt8812au card.
2. Try with channel 165 and svpcom/rtl8812au/v5.6.4.2 driver

**Reply by amtcvx** [reply]
Hello

Bravo you did the trick !

Now I have to look how these services (NetworkManager and wpa_supplicant) disturb silently the interface in reception on NVIDIA Jetpack using 88XXau, and have no impact on other configuration.

By the way, do you plan to implement 5.9.3.2 (aircrack-ng driver), that can overide 1Mbps (88XXau rtw_monitor_disable_1m=1) ?

Thank you 
XP

**Reply by svpcom** [reply]
Issues with NetworkManager and wpa_supplicant described in https://github.com/svpcom/wifibroadcast/wiki/Setup-HOWTO (section 7)

In 5.9.3.2 `rtw_monitor_disable_1m` is useless - it doesn't used anywere in the code:
```
$ git branch
* v5.7.0
$ git grep monitor_disable_1m
include/drv_types.h:    u8 monitor_disable_1m;
os_dep/linux/os_intfs.c:int rtw_monitor_disable_1m = 0;
os_dep/linux/os_intfs.c:module_param(rtw_monitor_disable_1m, int, 0644);
os_dep/linux/os_intfs.c:MODULE_PARM_DESC(rtw_monitor_disable_1m, "Disable default 1Mbps rate for monitor injected frames");
os_dep/linux/os_intfs.c:        registry_par->monitor_disable_1m = (u8)rtw_monitor_disable_1m;
```
